### PR TITLE
feat: added sqs info for orderbooks

### PR DESCRIPTION
**Pull request by crnbarr93**
## What is the purpose of the change:
These changes incorporate SQS data when interacting with on chain orderbooks.

Also includes a fix for market order inputs.

## Brief Changelog
- Added `queryOrderbookPools` query
- Altered `useOrderbooks` to use the above query
- Fixed market order spot price reference

### CODE REVIEW

File: packages/server/src/queries/complex/orderbooks/pools.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+export function getOrderbookPools() {
+  return cachified({
+    cache: orderbookPoolsCache,
+    key: `orderbookPools`,
+    ttl: 1000 * 60 * 60, // 1 hour
+    getFreshValue: () =>
+      queryPools().then((data) => {
+        return data
+          .filter((pool) =>
+            orderBookCodeIds.includes((pool as any).chain_model.code_id)
```
Comment (fabryscript) [inline packages/server/src/queries/complex/orderbooks/pools.ts:29]: Do we not have any typing for this yet?

File: packages/server/src/queries/complex/orderbooks/pools.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+          )
+          .map((pool) => {
+            return {
+              baseDenom: (pool as any).balances[1].denom,
+              quoteDenom: (pool as any).balances[0].denom,
```
Comment (fabryscript) [inline packages/server/src/queries/complex/orderbooks/pools.ts:34]: just a question, is by design the quote denom the first element and the base the second?

File: packages/web/components/place-limit-tool/index.tsx [hunk 3]
```diff
@@ -129,7 +129,11 @@ export const PlaceLimitTool: FunctionComponent<PlaceLimitToolProps> = observer(
               <span className="body2 text-osmoverse-300">
                 {swapState.baseAsset?.coinDenom}{" "}
                 {t("assets.table.price").toLowerCase()} ≈{" "}
-                {formatPretty(swapState.priceState.spotPrice ?? new Dec(0))}{" "}
+                {formatPretty(
+                  orderDirection === "bid"
```
Comment (fabryscript) [inline packages/web/components/place-limit-tool/index.tsx:133]: I might be wrong here, so you want to display the ask spot price when the direction is bid?

File: packages/web/hooks/limit-orders/use-orderbook.ts [hunk 4]
```diff
@@ -10,34 +10,12 @@ import { useSwapAsset } from "~/hooks/use-swap";
 import { useStore } from "~/stores";
 import { api } from "~/utils/trpc";
 
-interface Orderbook {
-  baseDenom: string;
-  quoteDenom: string;
-  contractAddress: string;
-}
-
 const USDC_DENOM = process.env.NEXT_PUBLIC_IS_TESTNET
```
Comment (fabryscript) [inline packages/web/hooks/limit-orders/use-orderbook.ts:13]: Aren't these the same minimal denoms?

**Review by fabryscript — CHANGES_REQUESTED** [review]

File: packages/server/src/queries/complex/orderbooks/pools.ts (hunk 2 above)
Comment (crnbarr93) [reply packages/server/src/queries/complex/orderbooks/pools.ts:34]: Yes, it's what's returned from the contract

File: packages/web/components/place-limit-tool/index.tsx (hunk 3 above)
Comment (crnbarr93) [reply packages/web/components/place-limit-tool/index.tsx:133]: A market order is run against the price of the opposite direction

File: packages/web/hooks/limit-orders/use-orderbook.ts (hunk 4 above)
Comment (crnbarr93) [reply packages/web/hooks/limit-orders/use-orderbook.ts:13]: Yea forgot to paste in the testnet denom again, good spot!

**Review by fabryscript — APPROVED** [review]

**Pull request merged**